client_integration_test.go

```diff
@@ -1233,6 +1234,19 @@ func TestClientWrite(t *testing.T) {
 		if got != tt.n {
 			t.Errorf("Write(%v): wrote: want: %v, got %v", tt.n, tt.n, got)
 		}
+
+		err = w.Sync()
+		switch err := err.(type) {
+		case nil:
+		case *StatusError:
+			// XXX We can't distinguish an actual fsync not supported
+			// error from a typo in the extension name here.
+			t.Logf("Sync: %v", err)
+			assert.Equal(t, ErrSSHFxOpUnsupported, err.FxCode())
```

Comment:
Yes, this is primarily so I don't have to write the server part. I only use the client. We could check whether the test is running against a real server and be stricter in that case; OpenSSH has had this extension since 2013.

Reply:
I would highly recommend against making the tests smarter than they need to be, like, checking to see if the server supports the feature or not, before running the test. You end up in a situation where the test is no longer clear, and “is this passing because it actually behaves as we expect it to, or because the server unexpected does not support the feature.”

I would recommend instead putting it in its own `TestClientSync` that will expect _only_ the `SSHFxOpUnsupported` case, and explicitly document with comments that this test will need to be reworked once the internal server actually supports the function. (Which the implementer should come across, because the test should actually fail at that point.)

Reply:
I've replaced this by two integration tests, including one for an actual server. I do like to check whether the method actually works with OpenSSH :)

Reply:
Great solution! Thanks for sticking through, sorry about the delays.